### ISSUE TITLE
WexBim image is faded when a detail is selected on the left column 

### ISSUE THREAD

**Issue by annastathaki** [xBimTeam/XbimWebUI#71]
Hi
I have just started using the WeXplorer with an ifc model of my own. Unfortunately, after dispaying the initial view correctly, fails to display the details of the Spacial Structure, Asset Types etc. Instead,it displays a faded version of the initial view.
Assuming that this had to do with my model, I used the Lakeside ifc sample from which I generated the wexbim format from the tools provided (XbimExhcange, XbimConvert in XbimUtilites or even the XbimXplorer in XbimWindows). Whatever tool I used to generate it, the file showed th same corrupted display when double clicking an any detail. 
Also, I should note that the original 'LakesideRestaurant.wexbim' file in XbimWebUI is 3.8 MB while the generated version is only 2.9 MB. I am showing a picture below:

![wexbim_issue](https://cloud.githubusercontent.com/assets/16819700/12483437/b503f790-c05d-11e5-99f4-0090faab6f7a.JPG)

I am clearly doing something wrong. Can you please advise me on how to correct this issue? 

Many thanks

**Comment by martin1cerny** [comment]
This is an intentional functionality called X-Ray mode. It shows everything semitransparent light blue except for the selected object which is highlighted. It is sometimes hard to find objects inside of the model and this helps a lot. 

However, this is not a default functionality but is set up this way in the sample application in development environment. Just click somewhere and it should change back to normal representation. When you do your own application it is up to you if you use X-Ray rendering mode or not. There are [other sample applications](https://github.com/xBimTeam/XbimWebUI/blob/master/Xbim.WeXplorer/simpleviewer.html) in the XbimWebUI project which are more simple and won't confuse you.

**Comment by annastathaki** [comment]
Hi
Thank you for your prompt reply. However, what I want to point out is that the image does not zoom in to show the selected item. With the original LakesideRestaurant.wexbim, the image does zoom in and the selected item is highlighted, while everything else is shown as you describe. With the wexbim file that I generated, as described in my original message, the image does not zoom in to show the selected item. It remains in its original view but in X-ray mode. See, for example, the item that is selected in the screenshot of my previous message. Using the original wexbim, the view I get is shown here below: 

![wexbim_issue1](https://cloud.githubusercontent.com/assets/16819700/12493723/0f88142c-c08f-11e5-90e8-e82ea2a026a4.JPG)

If you compare the two screenshots you can see what I am trying to describe. I hope this makes sense.

**Comment by martin1cerny** [comment]
You have to use wexBIM file and JSON file generated from the same file because it relies on entity label which is not stable between different versions of the same file (even two exports of the same model are not granted to keep entity labels the same).

**Comment by annastathaki** [comment]
Hi Martin, 
thank you for your help, I used to generate the json file using Xbim.CobiLightUK.Client from the Xbim.Exchange suite while the wexbim file was generated from XBim.Exchange and this did not work for some reason. I followed your advise and regenerated the json file from XBim.Exchange from which, however, I got a file with the name LakesideRestaurant_DpoW.json. I renamed it to LakesideRestaurant.json and it worked fine :).  I applied the procedure also to my own model and it worked too! 
One last query: What is the correct approach to generate the json file?
